src/org/opendatakit/briefcase/util/FormCache.java

```diff
@@ -1,51 +1,59 @@
 package org.opendatakit.briefcase.util;
 
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.FileOutputStream;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.exists;
+
 import java.io.IOException;
+import java.io.InputStream;
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import org.opendatakit.briefcase.model.BriefcaseFormDefinition;
+import org.opendatakit.briefcase.reused.BriefcaseException;
+import org.opendatakit.briefcase.reused.UncheckedFiles;
 
 public class FormCache implements FormCacheable {
-  private final File cacheFile;
-  private Map<String, String> pathToMd5Map = new HashMap<>();
-  private Map<String, BriefcaseFormDefinition> pathToDefinitionMap = new HashMap<>();
+  public static final String CACHE_FILE_NAME = "cache.ser";
+  private final Path cacheFile;
+  private final Map<String, String> pathToMd5Map;
+  private final Map<String, BriefcaseFormDefinition> pathToDefinitionMap;
 
-  @SuppressWarnings("unchecked")
-  public FormCache(File storagePath) {
-    cacheFile = new File(storagePath, "cache.ser");
-    if (cacheFile.exists() && cacheFile.canRead()) {
-      try (ObjectInputStream objectInputStream = new ObjectInputStream(new FileInputStream(cacheFile))) {
-        pathToMd5Map = (Map) objectInputStream.readObject();
-        pathToDefinitionMap = (Map) objectInputStream.readObject();
-      } catch (IOException | ClassNotFoundException | ClassCastException e) {
-        e.printStackTrace();
-      }
-    }
+  public FormCache(Path cacheFile, Map<String, String> pathToMd5Map, Map<String, BriefcaseFormDefinition> pathToDefinitionMap) {
+    this.cacheFile = cacheFile;
+    this.pathToMd5Map = pathToMd5Map;
+    this.pathToDefinitionMap = pathToDefinitionMap;
+    Runtime.getRuntime().addShutdownHook(new Thread(this::save));
+  }
 
-    Runtime.getRuntime().addShutdownHook(new Thread() {
-      @Override
-      public void run() {
-        save();
+  @SuppressWarnings("unchecked")
+  public static FormCache from(Path briefcaseDir) {
+    Path cacheFile = briefcaseDir.resolve(CACHE_FILE_NAME);
+    if (exists(cacheFile))
+      try (InputStream in = Files.newInputStream(cacheFile);
+           ObjectInputStream ois = new ObjectInputStream(in)) {
+        Map<String, String> pathToMd5Map = (Map<String, String>) ois.readObject();
+        Map<String, BriefcaseFormDefinition> pathToDefinitionMap = (Map<String, BriefcaseFormDefinition>) ois.readObject();
+        return new FormCache(cacheFile, pathToMd5Map, pathToDefinitionMap);
+      } catch (IOException | ClassNotFoundException e) {
+        throw new BriefcaseException("Can't deserialize form cache file", e);
```

Comment:
2018-05-24 11:17:43,765 [main] ERROR o.o.a.p.BaseFormParserForJavaRosa - Failed to launch GUI
org.opendatakit.briefcase.reused.BriefcaseException: Can't deserialize form cache file
    at org.opendatakit.briefcase.util.FormCache.from(FormCache.java:44)
Caused by: java.io.InvalidClassException: org.opendatakit.briefcase.model.BriefcaseFormDefinition; local class incompatible: stream classdesc serialVersionUID = 3811217215466839668, local class serialVersionUID = 3138267492033119752

Consider: Don’t throw the exception, ignore the incompatible cache file, and create a new one.